core/domain/activity_jobs_one_off_test.py

```diff
@@ -100,3 +115,119 @@ def mock_add_documents_to_index(docs, index):
+        ):
+            super(SnapshotMetadataModelsIndexesJobTest, self).setUp()
+
+    def _run_one_off_job(self):
+        """Runs the one-off MapReduce job."""
+        job_id = (
+            activity_jobs_one_off.SnapshotMetadataModelsIndexesJob.create_new())
+        activity_jobs_one_off.SnapshotMetadataModelsIndexesJob.enqueue(job_id)
+        self.assertEqual(
+            self.count_jobs_in_taskqueue(
+                taskqueue_services.QUEUE_NAME_ONE_OFF_JOBS), 1)
+        self.process_and_flush_pending_tasks()
+        stringified_output = (
+            activity_jobs_one_off.SnapshotMetadataModelsIndexesJob
+            .get_output(job_id))
+        eval_output = [ast.literal_eval(stringified_item)
+                       for stringified_item in stringified_output]
+        output = [(eval_item[0], int(eval_item[1]))
+                  for eval_item in eval_output]
+        return output
+
+    def _check_model_validity(self, original_model, migrated_model):
+        """Checks if the model was migrated correctly."""
+        self.assertEqual(
+            migrated_model.committer_id,
+            original_model.committer_id)
+        self.assertEqual(
+            migrated_model.commit_type,
+            original_model.commit_type)
+
+    def test_successful_migration_all_model_types(self):
+        for model_type in self.TESTED_MODEL_TYPES:
+            instance_id = 'id_1'
+            model = model_type(
+                id=instance_id,
+                committer_id='committer_id',
+                commit_type='create')
+            model.put()
+
+            output = self._run_one_off_job()
+            self.assertEqual(output, [(u'SUCCESS', 1)])
+            migrated_model = model_type.get_by_id(instance_id)
+            self._check_model_validity(model, migrated_model)
+
+            model.delete()
+
+    def test_successful_migration_multiple_models(self):
+        instance_id_1 = 'id_1'
+        instance_id_2 = 'id_2'
+        instance_id_3 = 'id_3'
+        collection_model = collection_models.CollectionSnapshotMetadataModel(
+            id=instance_id_1,
+            committer_id='committer_id',
+            commit_type='create')
+        collection_model.put()
+        exploration_model = exploration_models.ExplorationSnapshotMetadataModel(
+            id=instance_id_2,
+            committer_id='committer_id',
+            commit_type='create')
+        exploration_model.put()
+        skill_model = skill_models.SkillRightsSnapshotMetadataModel(
+            id=instance_id_3,
+            committer_id='committer_id',
+            commit_type='create')
+        skill_model.put()
+
+        output = self._run_one_off_job()
+        self.assertEqual(output, [(u'SUCCESS', 3)])
+
+        migrated_collection_model = (
+            collection_models.CollectionSnapshotMetadataModel.get_by_id(
+                instance_id_1))
+        self._check_model_validity(collection_model, migrated_collection_model)
```

Comment:
Include stronger checks for other, class-specific fields.

Reply:
Sorry, ignore this -- the child classes are trivial.

That said -- maybe include some checks for the other fields too like commit_message and commit_cmds?

Reply:
Done.